src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,143 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationFailureException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int MIN_VALID_AGE = 18;
+    private static final String VALID_LOGIN = "someLogin";
+    private static final String VALID_PASSWORD = "somePassword";
+    private static final String NULL_USER_MESSAGE
+            = "User registration failed due to user is null";
+    private static final String INVALID_AGE_MESSAGE
+            = "User registration failed due to user is null";
+    private static final String INVALID_LOGIN_PASSWORD_LENGTH_MESSAGE
+            = "User registration failed due to user is null";
+    private static final String USER_EXISTS_MESSAGE
+            = "User registration failed due to user is null";
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User user = new User();
+
+    @BeforeAll
+    static void setup() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    public void initializeValidUser() {
+        Storage.people.clear();
+        user.setAge(MIN_VALID_AGE);
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);

```

Comment:
to make sure each test has a separate user, because a shared link is not very good to have in tests
```suggestion
        user = new user();
        user.setAge(MIN_VALID_AGE);
        user.setLogin(VALID_LOGIN);
        user.setPassword(VALID_PASSWORD);
```